### PR TITLE
feat: exclude underscore from TS key enum rendering

### LINKED ISSUE #609: Exclude `_` from TypeScript key enum rendering
#### Reason/Context
Given an enum key such as `something_something` the enum renderer creates the following key `SOMETHING_UNDERSCORE_SOMETHING` even when TS fully supports `SOMETHING_SOMETHING`

#### Description

We need to add `_` to the exclusion list here: https://github.com/asyncapi/modelina/blob/71ac65a5aafd44731f7d421aec88c2ccfbfc2cad/src/generators/typescript/renderers/EnumRenderer.ts#L53

### PR DESCRIPTION
**Pull request by Ishan-Saini**
**Description**

- Added `_` to the exclusion list inside TS key enum renderer.
- Added tests for EnumRenderer.

**Related issue(s)**
Resolves #609

### CODE REVIEW

**Comment by Ishan-Saini** [comment]
Hey @jonaslagoni , I am not sure if this falls under the `chore:` specification.
Also, let me know if something needs to be changed.

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
> Hey @jonaslagoni , I am not sure if this falls under the chore: specification.
Also, let me know if something needs to be changed.

You can use `feat` here 🙂 

I can see we did not create a test for the EnumRenderer... Do you mind creating a test file for it under `test/generators/typescript/renderers/EnumRenderer.spec.ts`? 🙏 

For test cases, all you need to test is `normalizeKey`:
- should correctly convert underscore character. 
  - Given `something_something` you expect `SOMETHING_SOMETHING` to be returned.
- should correctly convert space character.
  - Given `something something` you expect `SOMETHING_SOMETHING` to be returned.

Feel free to add more cases if you feel like it 🙂

**Comment by Ishan-Saini** [comment]
> I can see we did not create a test for the EnumRenderer... Do you mind creating a test file for it under `test/generators/typescript/renderers/EnumRenderer.spec.ts`? pray

I am new to testing but I'd still try to get it done asap. 😅

**Comment by jonaslagoni** [comment]
> I am new to testing but I'd still try to get it done asap.

No worries!

I can give you the core testing setup, that way you only need to add your own tests, let me know if you want that 🙂

Otherwise ping me if you get stuck 👍

**Comment by Ishan-Saini** [comment]
> I can give you the core testing setup if you want that, that way you only need to add the tests for the function, let me know slightly_smiling_face

I'll try to get it done myself first.
Will let you know if I get stuck :)

**Comment by Ishan-Saini** [comment]
> I can give you the core testing setup if you want that, that way you only need to add the tests for the function, let me know slightly_smiling_face
Hey @jonaslagoni , Could you please provide the core testing setup 😬? 
A bit stuck, I think the setup would help me get started.

**Comment by jonaslagoni** [comment]
> A bit stuck, I think the setup would help me get started.

For sure!

Here you go @Ishan-Saini (`test/generators/typescript/renderers/EnumRenderer.spec.ts`):
```ts
import { TypeScriptGenerator } from '../../../../src/generators';
import { EnumRenderer } from '../../../../src/generators/typescript/renderers/EnumRenderer';
import { CommonInputModel, CommonModel } from '../../../../src/models';

describe('TypeScriptRenderer', () => {
  let renderer: EnumRenderer;
  beforeEach(() => {
    renderer = new EnumRenderer(TypeScriptGenerator.defaultOptions, new TypeScriptGenerator(), [], new CommonModel(), new CommonInputModel());
  });

  describe('normalizeKey()', () => {
    test('should correctly format " " to correct key', () => {
      
    });
    test('should correctly format "_" to correct key', () => {
      
    });
  });
});

```

File: test/generators/typescript/renderers/EnumRenderer.spec.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { TypeScriptGenerator } from '../../../../src/generators';
+import { EnumRenderer } from '../../../../src/generators/typescript/renderers/EnumRenderer';
+import { CommonInputModel, CommonModel } from '../../../../src/models';
+
+describe('TypeScriptRenderer', () => {
```
Comment (jonaslagoni) [inline test/generators/typescript/renderers/EnumRenderer.spec.ts:5]: Ups, forgot to change `TypeScriptRenderer` to `EnumRenderer` 😅

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Small change. 

Also, I see you updated `package-lock.json`, do you mind reverting it to the current one? Just [copy-paste it from master](https://github.com/asyncapi/modelina/blob/master/package-lock.json)and commit it 🙂 

It is a known "problem" with never versions of node, but our CI uses older versions, so we need to keep it to their liking ([for now](https://github.com/asyncapi/.github/issues/123)).

File: test/generators/typescript/renderers/EnumRenderer.spec.ts (hunk 1 above)
Comment (Ishan-Saini) [reply test/generators/typescript/renderers/EnumRenderer.spec.ts:5]: Yep, I did have a thought about it 😬

**Comment by Ishan-Saini** [comment]
> Also, I see you updated `package-lock.json`, do you mind reverting it to the current one? Just [copy-paste it from master](https://github.com/asyncapi/modelina/blob/master/package-lock.json)and commit it slightly_smiling_face
> 
> It is a known "problem" with never versions of node, but our CI uses older versions, so we need to keep it to their liking ([for now](https://github.com/asyncapi/.github/issues/123)).

Oh okay, I will keep that in mind :)

File: test/generators/typescript/renderers/EnumRenderer.spec.ts (hunk 1 above)
Comment (jonaslagoni) [reply test/generators/typescript/renderers/EnumRenderer.spec.ts:5]: Haha, trust yourself when you notice something like that (regardless of who gives you the code) and raise the flag 👍

Comment (Ishan-Saini) [reply test/generators/typescript/renderers/EnumRenderer.spec.ts:5]: Aye aye captain!

Comment (jonaslagoni) [reply test/generators/typescript/renderers/EnumRenderer.spec.ts:5]: @Ishan-Saini I dont see the change to the test name? DId I miss the commit or?

Comment (Ishan-Saini) [reply test/generators/typescript/renderers/EnumRenderer.spec.ts]: Oops, my apologies. I pushed the changes.
It was a silly mistake.

**Review by jonaslagoni — APPROVED** [review]
LGTM 🎉

**Pull request merged**